グループ機能実装

hidetoshi1230 opened the pull request:
#WHAT 
グループ機能実装

#WHY 
サービスに必要なため。

hidetoshi1230 committed:
グループ機能実装１（コントローラ、モデル作成）

・groupsコントローラ、モデルを作成
・テーブルにカラム追加

hidetoshi1230 committed:
グループ機能実装２（ルーティング～ビュー設置）

・groupのルーティングを設定する
・groupコントローラのアクションを定義する
・アクセス先のビューを設置（haml変換）し、リンクするようにする。

yamamotokeishiro commented on config/routes.rb:
```diff
@@ -2,4 +2,5 @@
   devise_for :users
   root "messages#index"
   resource :user, only: [:edit, :update]
+  resource :group
```
onlyを使用して、今回使用するアクションだけ指定してください！

hidetoshi1230 committed:
グループ機能実装３（修正）

onlyを使用して、今回使用するアクションだけ指定してください！

yamamotokeishiro commented on config/routes.rb:
```diff
@@ -2,5 +2,5 @@
   root "messages#index"
   resource :user, only: [:edit, :update]
-  resource :group
+  resource :group, only: [:new, :create, :edit, :update]
```
indexも必要かと！

hidetoshi1230 committed:
グループ機能実装４（修正）

onlyを使用して、今回使用するアクションだけ指定してください！

hidetoshi1230 committed:
グループ機能実装５（修正）

onlyを使用して、今回使用するアクションだけ指定してください！

hidetoshi1230 committed:
グループ機能実装６（グループ作成）

グループ作成機能を実装する
form_forでのhaml書き換え
collection_check_boxesの実装
フラッシュメッセージの実装
エラーメッセージの実装
リダイレクト設定

yamamotokeishiro commented on app/controllers/groups_controller.rb:
```diff
@@ -1,9 +1,16 @@
+    if @group.save
+      redirect_to root_path, notice: "グループを作成しました。"
+    else
+      render :new
```
失敗した際の処理も記述してください！

hidetoshi1230 committed:
グループ機能実装７（修正）

失敗した際の処理も記述してください！

hidetoshi1230 committed:
グループ機能実装８（作成したグループを表示）

ログインしているユーザーの作成したグループを表示する

hidetoshi1230 committed:
グループ機能実装９（修正）

コントローラ修正

hidetoshi1230 committed:
グループ機能実装（グループ編集機能を実装する）

グループ編集機能を実装する
・編集画面の実装
・hamlの切り分け
・フラッシュメッセージ実装

hidollara commented on app/views/layouts/_group_new.html.haml:
```diff
@@ -0,0 +1,45 @@
+    .chat-group-form__field.clearfix
+      .chat-group-form__field--left
+      .chat-group-form__field--right
+        = f.submit "登録する", class: "chat-group-form__action-btn"
```
_group_new.hamlと_group_edit.hamlはほとんど同じ内容ですね。統合できませんか？

hidetoshi1230 committed:
グループ機能実装11（修正）

_group_new.hamlと_group_edit.hamlはほとんど同じ内容ですね。統合できませんか？

ken-nag commented:
lgtm

hidetoshi1230 committed:
グループ機能実装12（ルートパス変更）

ルートパスを変更する

kouuuki commented:
LGTMです！
